common/frame.go

```diff
@@ -315,27 +315,51 @@ func (f *Frame) defaultTimeout() time.Duration {
 func (f *Frame) document() (*ElementHandle, error) {
 	f.log.Debugf("Frame:document", "fid:%s furl:%q", f.ID(), f.URL())
 
-	rt := k6common.GetRuntime(f.ctx)
-	if f.documentHandle != nil {
-		return f.documentHandle, nil
+	if cdh, ok := f.cachedDocumentHandle(); ok {
+		return cdh, nil
```

Comment:
`ok` is used for minimizing the impact of data races in the `document` method if the `documentHandle` is not cached. If it is cached it returns it. There is still a tiny window of race condition (_not data race_) here: We may miss the cache and reacquire the document handle. That race condition can even be a good thing because it'll get a fresh one!